nengo/params.py

```diff
@@ -51,8 +51,20 @@ class Parameter(object):
     readonly : bool, optional
         If true, the parameter can only be set once.
         By default, parameters can be set multiple times.
+
+    Attributes
+    ----------
+    coerce_defaults : bool (Default: True)
+        If True, validate default values for this parameter if they are changed
+        by the config system. Setting a value on a parameter object instance
+        will always be validated.
```

Comment:
Would it be more accurate to say 'when' instead of 'if' in the first sentence? I.e.: “If True, validate default values for this parameter **when** they are changed by the config system.” I would understand that as that this validation happens whenever the value is changed (by the config system), whereas the variant with 'if’ seems to imply that the validation could be separate from changing the value as long as the value is changed at all. But I'm not a native speaker, so maybe the 'if' works equally well.

In the second sentence: What does 'parameter object instance' exactly mean? Object and instance are often the same thing. Could we do with just one of the words? And what exactly is the distinction to the case described in the first sentence? Is it setting a non-default value (I assume?) or is it setting a (potentially default value) on a `Parameter` instance (as opposed to the `Parameter` class)?

Reply:
This is trying to distinguish between (1) `config[obj].attr = val` versus (2) `obj.attr = val`.  The user may not have any `config[obj].attr = val` lines in their code (i.e., they never change the defaults), in which case `coerce` would never be called in the (1) format and it wouldn't matter if `coerce_defaults` was True or False (hence the "if").

The second sentence is trying to describe (2) (`obj.attr = val`).  It might be clearer just to take that part out, since it's describing behaviour that isn't really connected to this parameter (I was just worried that people would think there was no validation being performed if `coerce_defaults=False`).

Reply:
Thanks, I understand now better what you intended to say. How about:

> If True, validate values for this parameter when they are set in a Config object. Setting a parameter directly on an object will always be validated.

Reply:
Sounds good